test/AutoRest.TestServerLowLevel.Tests/dpg-customization.cs

```diff
@@ -74,6 +74,61 @@ await foreach (Page<BinaryData> page in allPages.AsPages())
             }
         });
 
+        [Test]
+        public Task HandwrittenModelLro() => Test(async (host) =>
+        {
+            using var diagnosticListener = new ClientDiagnosticListener("dpg_customization_LowLevel", asyncLocal: true);
+            CollectionAssert.IsEmpty(diagnosticListener.Scopes);
+
+            Operation<Product> lro = await new DPGClient(Key, host).LroValueAsync(WaitUntil.Started, "model");
+            Assert.AreEqual(1, diagnosticListener.Scopes.Count);
+            Assert.AreEqual("DPGClient.LroValue", diagnosticListener.Scopes[0].Name);
+            Assert.IsTrue(diagnosticListener.Scopes[0].IsCompleted);
```

Comment:
You can use [AssertAndRemoveScope](https://github.com/Azure/azure-sdk-for-net/blob/72fc7b6182f15d46a8afb529af2f5fb6730c4133/sdk/core/Azure.Core.TestFramework/src/ClientDiagnosticListener.cs#L201) which will throw if the scope is not completed and allow you to specify a scope name and attributes that the scope should contain.

Reply:
Fixed